text/0000-road-to-typescript.md

```diff
@@ -0,0 +1,160 @@
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Ember Data, Ember CLI, Learning, Steering
+RFC PR: 
+---
+
+<!--- 
+Directions for above: 
+
+Stage: Leave as is
+Start Date: Fill in with today's date, YYYY-MM-DD
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Fill this in with the [team(s)](README.md#relevant-teams) to which this RFC applies
+RFC PR: Fill this in with the URL for the Proposal RFC PR
+-->
+
+# Road to TypeScript
+
+## Summary
+
+TypeScript has become a key part of the front-end development ecosystem over the past several years, and powers many of the best developer experiences in the front-end ecosystem. Ember was a relatively early TypeScript adopter for its internals, and there is widespread usage in the ecosystem with community support, but to date Ember has not provided “out of the box” or official support for TypeScript. This RFC declares our intent to make TypeScript a first-class citizen of the Ember ecosystem, as a peer to JavaScript which makes the developer experience better for *all* Ember developers. It outlines the key constraints and goals for the effort, and outlines a roadmap for accomplishing those goals.
+
+## Motivation
+
+TypeScript is a key ingredient of contemporary front-end development. Per community surveys like the GitHub Octoverse, the Stack Overflow Developer report, and the State of JS survey, TypeScript has exploded in popularity over the last half decade, and is the rare tool which has continuously grown in both both usage and satisfaction over that time. ( While all of these surveys should be understood to be wildly unrepresentative of the broader front-end development community in various ways, they nonetheless provide some useful signal about the state of the ecosystem.)
+
+Additionally, the development of the language server protocol and the widespread adoption of the TypeScript language server has led to first-class support for TypeScript across many tools, from Vim to Visual Studio. This support also powers rich developer experiences for JavaScript-only developers, as it can provide smart autocomplete, inline documentation, and more even for codebases which are written solely in JavaScript. Thus, non-TypeScript and TypeScript users alike benefit when working with a project which supplies type definitions.
+
+Ember users have been working with TypeScript for years with community support led by the Typed Ember team, but there is one critical problem which cannot be solved without making TypeScript support official, and several key ways in which official support would substantially improve the experience of both TypeScript and JavaScript developers:
+
+- [ ] the SemVer problem
+    - DefinitelyTyped
+    - TS version support policy
+- [ ] other key benefits for official adoption
+    - coordination (e.g. editions aren't done till docs also goes for TS and types)
+    - [ ] blueprints?
+    - [ ] more?
```

Comment:
a more immersive experience for TS user from the ember (cli) tools - such as the generators

Reply:
Integrated in 3843459.